src/transformers/modeling_tf_utils.py

```diff
@@ -1978,16 +2319,26 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         else:
             model(model.dummy_inputs)  # build the network with dummy inputs
 
-        assert os.path.isfile(resolved_archive_file), f"Error retrieving file {resolved_archive_file}"
         # 'by_name' allow us to do transfer learning by skipping/adding layers
         # see https://github.com/tensorflow/tensorflow/blob/00fad90125b18b80fe054de1055770cfb8fe4ba3/tensorflow/python/keras/engine/network.py#L1339-L1357
         try:
-            missing_keys, unexpected_keys, mismatched_keys = load_tf_weights(
-                model,
-                resolved_archive_file,
-                ignore_mismatched_sizes=ignore_mismatched_sizes,
-                _prefix=load_weight_prefix,
-            )
+            if is_sharded:
+                for file in resolved_archive_file:
+                    assert os.path.isfile(file), f"Error retrieving files {file}"
```

Comment:
No new asserts in the codebase ;-) Please use a test and raise the appropriate error.

Reply:
Error is pretty much already handled with the `OSError` as the call to `load_tf_...` is already in a `try/catch` clause

Reply:
In this case you can remove the assert :-p